lib/experimental/html/class-wp-html-walker.php

```diff
@@ -568,7 +571,7 @@ public function is_closed() {
 	public function add_class( $class_name ) {
 		$this->assert_not_closed();
 		if ( $this->tag_name ) {
-			$this->classnames_updates[ self::comparable( $class_name ) ] = new WP_HTML_Class_Name_Update( $class_name, true );
+			$this->classname_updates[ $class_name ] = self::ADD_CLASS;
```

Comment:
Just wanted to comment about the comment from @dmsnell in the private message. [class names are case-sensitive ](https://stackoverflow.com/a/12533957). I believe that's why we don't want to use here `self:comparable`. It also trims whitespace, does it mean we don't care about that part here anymore? It would be great to add a test case that uses excessive spacing.

Reply:
shouldn't matter here because we're adding a class. we might consider it when removing, but I think it's okay to let things break if someone tries to remove the class `"  something"` - that's not a reasonable thing to expect to happen, and DOM implementations will throw if you try to remove a class with spaces in it